sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -128,6 +128,16 @@ object ScalaReflection extends ScalaReflection {
     case _ => false
   }
 
+  def isValueClass(tpe: `Type`): Boolean = {
+    tpe.typeSymbol.asClass.isDerivedValueClass
+  }
+
+  /** Returns the underlying type of value class `cls`. */
+  def getUnderlyingTypeOf(cls: `Type`): `Type` = {
```

Comment:
nit: it is usually to name it as `tpe` for `Type` in ScalaReflection.

Reply:
agree, it's better to use `tpe`